smoketest/utils.py

```diff
@@ -38,20 +41,42 @@ def transform_url(url, scheme=None, port=None, level=None, cachebust=None):
 
 
 def level_transform(parts, level):
-    """Transform the first subdomain according to our habits.
+    """Transform the first subdomain according to our habits, normally.
     www.usnews.com => www-level.usnews.com
-    """
-    if level == 'live':
-        return parts
 
-    try:
-        subdomain, subdomain2, remaining = parts[1].split('.', 2)
-    except ValueError:
-        # no subdomain to modify
+    Can also be used for a more general replacements, see the docs:
+    {LEVEL}www.usnews.com => level-www.usnews.com
+    """
+    level_token = get_level_token()
+
+    if level_token and level_token in parts[1] + parts[2]:
+        host, path = parts[1], parts[2]
+        if level == 'live':
+            level = ''
+
+        if host:
+            host = host.replace(level_token, level)
+            host = host.replace('..', '.').replace('-.', '.')
+            if host[0] in ('-', '.'):
+                host = host[1:]
+        if path:
+            if level == '' and level_token + '/' in path:
+                path = path.replace(level_token+'/', level)
```

Comment:
Do you want to add spaces around this + too, like in bd8792d? ;)

Reply:
Yep, fixed. Thanks.